### PR TITLE
Expose Excel Style and Advanced filtering components

### LINKED ISSUE #5878: Container instance for igxGrid overlay components
## Is your feature request related to a problem? Please describe.
Current filter dialog could be instanced inside the grid only and that could be an issue on specific grid size.

## Describe the solution you'd like
Filter dialog to instanced in separate containers outside the grid.

## Describe alternatives you've considered
Create alternative UI and hook to all grid API methods to achieve the same behavior.

## Additional context
We would like to instantiate the IgxGrid Excel Style Filter and advanced filter dialog in a separate container, like for example, a flyout menu.

### PR DESCRIPTION
**Pull request by DiyanDimitrov**
Closes #5878

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [x] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 1]
```diff
@@ -5880,6 +5925,45 @@ describe('IgxGrid - Filtering actions - Excel style filtering #grid', () => {
...
+            grid = fix.componentInstance.grid;
+            fix.detectChanges();
+        }));
+
+        it('Should allow hosting Excel Style filtering component outside of the grid.', fakeAsync(() => {
+            // sort
+            const excelMenu = fix.nativeElement.querySelector('.igx-excel-filter__menu');
+            const sortComponent = excelMenu.querySelector('.igx-excel-filter__sort');
+            const sortAsc = sortComponent.lastElementChild.children[0].children[0];
+            sortAsc.click();
```
Comment (tachojelev) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts]: Click logic can be replaced with helper method: **clickSortAscInExcelStyleFiltering**

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 2]
```diff
@@ -5880,6 +5925,45 @@ describe('IgxGrid - Filtering actions - Excel style filtering #grid', () => {
...
+            fix.detectChanges();
+            expect(grid.sortingExpressions[0].fieldName).toEqual('ProductName');
+            expect(grid.sortingExpressions[0].dir).toEqual(SortingDirection.Asc);
+
+            // pin
+            const pinComponent = excelMenu.querySelector('.igx-excel-filter__actions-pin');
+            pinComponent.click();
```
Comment (tachojelev) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts]: Click logic can be replaced with helper method: **clickPinIconInExcelStyleFiltering**

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 3]
```diff
@@ -5880,6 +5925,45 @@ describe('IgxGrid - Filtering actions - Excel style filtering #grid', () => {
...
+            // filter
+            verifyExcelStyleFilterAvailableOptions(fix, [ 'Select All', '(Blanks)', 'Ignite UI for Angular',
+                'Ignite UI for JavaScript', 'NetAdvantage', 'Some other item with Script' ],
+                [ true, true, true, true, true, true ]);
+            toggleExcelStyleFilteringItems(fix, fix, true, 1, 4);
+            expect(grid.rowList.length).toBe(3);
+
+            // hide
+            const hideComponent = excelMenu.querySelector('.igx-excel-filter__actions-hide');
+            hideComponent.click();
```
Comment (tachojelev) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts]: Click logic can be replaced with helper method: **clickHideIconInExcelStyleFiltering**

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,7 @@ All notable changes for each version of this project will be documented in this
 
 - `IgxGrid`, `IgxTreeGrid`, `IgxHierarchicalGrid`
     - **Behavioral Change** - Pinning columns is no longer automatically prevented when the pinning area would exceed the size of the grid.
+- `IgxGridExcelStyleFilteringComponent` and `IgxAdvancedFilteringDialogComponent` can now be hosted outside of the grid in order to provide the same experience as the built-in filtering UI.
```
Comment (tachojelev) [inline CHANGELOG.md]: Maybe we can present this bullet point as a new feature in the **New Features** section.

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 5]
```diff
@@ -345,6 +345,12 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
     @Output()
     public filteringExpressionsTreeChange = new EventEmitter<IFilteringExpressionsTree>();
 
+    /**
+     *@hidden
+     */
+    @Output()
+    public advancedFilteringExpressionsTreeChange = new EventEmitter<IFilteringExpressionsTree>();
```
Comment (tachojelev) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts]: We should remove the **@hidden** tag and document this emitter as a **new** feature.

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts (hunk 1 above)
Comment (DiyanDimitrov) [reply projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts]: Done.

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts (hunk 2 above)
Comment (DiyanDimitrov) [reply projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts]: Done.

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts (hunk 3 above)
Comment (DiyanDimitrov) [reply projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts]: Done.

File: CHANGELOG.md (hunk 4 above)
Comment (DiyanDimitrov) [reply CHANGELOG.md]: Done.

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts (hunk 5 above)
Comment (DiyanDimitrov) [reply projects/igniteui-angular/src/lib/grids/grid-base.directive.ts]: Done.

Comment (DiyanDimitrov) [reply projects/igniteui-angular/src/lib/grids/grid-base.directive.ts]: Done.

**Pull request merged**